static/app/stores/tooltipStore.tsx

```diff
@@ -1,75 +1,79 @@
-type TooltipStoreInterface = {
-  addTooltip(tooltip: React.Component): void;
-  closeAllTooltips(): void;
-  getOpenableSingleTooltips(): React.Component[];
-  init(): TooltipStoreInterface;
-  openAllTooltips(): boolean;
-  removeTooltip(tooltip: React.Component): void;
-  tooltips: React.Component[];
+type TooltipActions = {
+  setOpen: (isOpen: boolean) => void;
+  setUseGlobalPortal: (useGlobalPortal: boolean) => void;
+};
+
+type Tooltip = {
+  actions: TooltipActions;
+  id: string;
 };
 
 const MAX_TOOLTIPS_TO_OPEN = 100;
 
-const TooltipStore: TooltipStoreInterface = {
-  tooltips: [],
+/**
+ * XXX: This is NOT a reflux store.
+ *
+ * This is purely used for acceptance tests where we want to open all tooltips
+ * in the DOM at once for capturing visual snapshots of tooltips being open.
+ */
+class TooltipStore {
+  tooltips = new Map<string, Tooltip>();
+
+  constructor() {
+    window.__openAllTooltips = this.openAllTooltips;
+    window.__closeAllTooltips = this.closeAllTooltips;
+  }
 
   getOpenableSingleTooltips() {
-    return this.tooltips.filter(tooltip => {
-      // Filtering out disabled tooltips and lists of tooltips (which cause rendering issues for snapshots) using the internal 'key'
-      const _internals =
-        (tooltip as any)._reactInternalFiber || (tooltip as any)._reactInternals;
-      return (
-        !(tooltip.props as any).disabled &&
-        !_internals.key &&
-        !(tooltip.props as any).disableForVisualTest
-      );
-    });
-  },
+    // TODO: We need to figure out how to find "list" tooltips by looking for
+    // the `key` like we did before. This was previously done with
+    //
+    //  const _internals =
+    //    (tooltip as any)._reactInternalFiber || (tooltip as any)._reactInternals;
```

Comment:
@k-fish Any idea how to figure this out in functional component land?

This is the only piece I am missing from this.